pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -690,24 +690,37 @@ public void updateBooleanFieldsIfNeeded(Schema oldSchema) {
 
   /**
    * Check whether the current schema is backward compatible with oldSchema.
-   * Backward compatibility requires all columns and fieldSpec in oldSchema should be retained.
+   * Always return true if force to set to true.
+   *
+   * Backward compatibility requires
+   * (1) all columns in oldSchema should be retained.
+   * (2) all column fieldSpecs should be backward compatible with the old ones.
    *
    * @param oldSchema old schema
+   * @param force whether to force overriding the old schema
    */
-  public boolean isBackwardCompatibleWith(Schema oldSchema) {
+  public boolean isBackwardCompatibleWith(Schema oldSchema, boolean force) {
```

Comment:
We should not put `force` into this method. The `force` should be handled on the caller side based on the return result of this method.

Reply:
Thanks, can you check the third commit?